src/Motor.Extensions.Diagnostics.Metrics/AutoObserveStopwatch.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Diagnostics;
+using System.Linq;
+using Prometheus.Client;
+
+namespace Motor.Extensions.Diagnostics.Metrics
+{
+    public class AutoObserveStopwatch : IDisposable
+    {
+        private readonly Stopwatch _watch;
+        private readonly ISummary? _unlabeledSummary;
+        private readonly IMetricFamily<ISummary>? _labeledSummary;
+        private readonly AbstractLabel[] _labels = Array.Empty<AbstractLabel>();
+
+        public AutoObserveStopwatch(ISummary? unlabeledSummary)
+        {
+            _unlabeledSummary = unlabeledSummary;
+            _watch = Stopwatch.StartNew();
+        }
+
+        public AutoObserveStopwatch(IMetricFamily<ISummary>? labeledSummary, params AbstractLabel[] labels)
+        {
+            _labeledSummary = labeledSummary;
+            _labels = labels;
+            _watch = Stopwatch.StartNew();
+        }
+
+        public void Dispose()
+        {
+            _watch.Stop();
+            _labeledSummary?.WithLabels(_labels.Select(l => l.ToString()).ToArray()).Observe(_watch.ElapsedMilliseconds);
```

Comment:
I'm wondering why you are applying the labels here instead of the constructor? Do you want to achieve lazy evaluation of the labels during the observation?

Reply:
Yes, I also tried to show this behavior in the example. The idea is, that e.g. for a long running check which returns `true`/`false`, one could want to measure the runtime for both cases independently using labels of the summary. These labels then only exist once the measured scope is completed.